File: Data Structures & Algorithms/max-consecutive-ones/submission-0.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int ones = 0;
        int maxOnes = 0;
        for (int n : nums) {
            if (n == 0) {
                maxOnes = Math.max(maxOnes, ones);
                ones = 0;
            } else {
                ones++;
            }
        }
        maxOnes = Math.max(maxOnes, ones);

        return maxOnes;
    }
}

File: Data Structures & Algorithms/longest-strictly-increasing-or-strictly-decreasing-subarray/submission-0.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        prev = nums[0]
        maxCount = 0
        count = 1

        for i in range(1, len(nums)):
            if nums[i] > prev:
                count += 1
            else:
                maxCount = max(maxCount, count)
                count = 1
            prev = nums[i]

        maxCount = max(maxCount, count)

        prev = nums[0]
        count = 1

        for i in range(1, len(nums)):
            if nums[i] < prev:
                count += 1
            else:
                maxCount = max(maxCount, count)
                count = 1
            prev = nums[i]
        
        maxCount = max(maxCount, count)


        return maxCount

